rholang/src/main/scala/coop/rchain/rholang/interpreter/dispatch.scala

```diff
@@ -109,20 +110,17 @@ class RholangAndScalaDispatcher[M[_]] private (
 
 object RholangAndScalaDispatcher {
 
-  def create[M[_], F[_]](
-      tuplespace: RhoISpace,
-      dispatchTable: => Map[Long, Function1[Seq[ListChannelWithRandom], M[Unit]]],
-      urnMap: Map[String, Par])(implicit
-                                parallel: Parallel[M, F],
-                                s: Sync[M],
-                                ft: FunctorTell[M, Throwable])
-    : (Dispatch[M, ListChannelWithRandom, TaggedContinuation], Reduce[M]) = {
-    val pureSpace          = PureRSpace[M].of(tuplespace)
+  def create(tuplespace: RhoISpace,
```

Comment:
```
 Fix `F[_]` to monix Task during creation of RholangAndScalaDispatcher.

This is necessary because Registry if fixed on Task and it seemed to me that
refactoring it to arbitrary effect would be much more time consuming. We never
really call RholangAndScalaDispatcher with anything different than Task anyway.
```

I fail to see how this is necessary (backed by 'seemed'). "It's only called with Task" is not a valid argument, for testability and maintainability reasons. If regarded a valid argument, it would justify throwing away all the tagless final effort.

Reply:
Given that 'Registry is fixed on task', I also see that we've both done a very poor job when reviewing the Registry PR. I can only excuse us by the size of the PR - which itself was inexcusable :)

Reply:
Instead of fixing Dispatcher to Task, please make the Registry abstract away from it. Change things for the better, not the worse. If you need that, I can do the Registry refactor for you.

Reply:
I explained it in the commit description. Abstracting away `Registry`  from the `Task` is a time consuming change. We never call `RholangWithScalaDispatcher` in tests because it is not easily testable. That's way fixing it on `Task` is fine.

Reply:
The purpose of this PR is not to change the implementation of the `Registry`.

Reply:
https://github.com/rchain/rchain/pull/1510

Reply:
Given #1510 is merged, there's no excuse for fixing over task. Please edit this commit so that it only touches the name registration part, and change the commit message. Notice also there's a likely merge conflict race between this and #1503 :) Which again reinforces the benefits of fine-grained commits.